packages/calypso-e2e/src/lib/test-account.ts

```diff
@@ -0,0 +1,161 @@
+import fs from 'fs/promises';
+import path from 'path';
+import chalk from 'chalk';
+import { BrowserContext, Page } from 'playwright';
+import { getAccountCredential, getAccountSiteURL, getCalypsoURL, config } from '../data-helper';
+import { TOTPClient } from '../totp-client';
+import { LoginPage } from './pages/login-page';
+
+/**
+ * Represents the WPCOM test account.
+ */
+export class TestAccount {
+	readonly accountName: string;
+	readonly credentials: [ string, string ];
+
+	/**
+	 * Constructs an instance of the TestAccount for the given account name.
+	 * Available test accounts should be defined in the e2e config file.
+	 */
+	constructor( accountName: string ) {
+		this.accountName = accountName;
+		this.credentials = getAccountCredential( accountName );
+	}
+
+	/**
+	 * Authenticates the account using previously saved cookies or via the login
+	 * page UI if cookies are unavailable.
+	 */
+	async authenticate( page: Page ): Promise< void > {
+		const browserContext = await page.context();
+		await browserContext.clearCookies();
+
+		if ( await this.hasFreshAuthCookies() ) {
+			this.log( 'Found fresh cookies, skipping log in' );
+			await browserContext.addCookies( await this.getAuthCookies() );
+			await page.goto( getCalypsoURL( '/' ) );
+		} else {
+			this.log( 'Logging in via Login Page' );
+			await this.logInViaLoginPage( page );
+		}
+	}
+
+	/**
+	 * Logs in via the login page UI. The verification code will be submitted
+	 * automatically if it's defined in the config file.
+	 */
+	async logInViaLoginPage( page: Page ): Promise< void > {
+		const loginPage = new LoginPage( page );
+
+		await loginPage.visit();
+		await loginPage.logInWithCredentials( ...this.credentials );
+
+		const verificationCode = this.getTOTP();
+		if ( verificationCode ) {
+			await loginPage.submitVerificationCode( verificationCode );
+		}
+	}
+
+	/**
+	 * Retrieves the Time-based One-Time Password (a.k.a. verification code) from
+	 * the config file if defined for the current account.
+	 */
+	getTOTP(): string | undefined {
+		const configKey = `${ this.accountName }TOTP`;
+		if ( ! config.has( configKey ) ) {
+			return undefined;
+		}
+
+		const totpClient = new TOTPClient( config.get( configKey ) );
+		return totpClient.getToken();
+	}
+
+	/**
+	 * Retrieves the site URL from the config file if defined for the current
+	 * account.
+	 *
+	 * @throws If the site URL is not available.
+	 */
+	getSiteURL( { protocol = true }: { protocol?: boolean } = {} ): string {
+		return getAccountSiteURL( this.accountName, { protocol } );
+	}
+
+	/**
+	 * Checks whether the current account has fresh auth cookies file available or
+	 * not. Authentication cookies are considered to be fresh when they have less
+	 * than 3 days.
+	 */
+	async hasFreshAuthCookies(): Promise< boolean > {
+		try {
+			const { birthtimeMs } = await fs.stat( this.getAuthCookiesPath() );
+			const nowMs = Date.now();
+			const threeDaysMs = 3 * 24 * 60 * 60 * 1000;
```

Comment:
I'm getting conflicting information on this one, maybe you can chime in with your experience.

I've had cookies be rejected at both 1 and 3 days. To be safe, I think it might be good idea to have the expiry date set to 1 day, but I'd like to base any decisions off an actual value as implemented in the backend if possible.

Reply:
Hmm, I wasn't able to test that on my end (can't seem to get the cookies rejected), but this might need some deeper look. I'd move that to a follow-up PR. To be clear, are you able to pre-authenticate with freshly created cookies though?

Reply:
> Hmm, I wasn't able to test that on my end (can't seem to get the cookies rejected), but this might need some deeper look. I'd move that to a follow-up PR. To be clear, are you able to pre-authenticate with freshly created cookies though?

To clarify, this was an observation of mine dating back to the early days of implementing pre-authenticated cookies. I haven't tested your changes locally, but have no doubts that it works! No changes are required for this PR.

This most recently came up in December (prior to my AFK) when I was looking at preserving cookies for the P2 E2E user to work around [this issue](https://github.com/Automattic/wp-calypso/issues/59211), but I kept having my cookies rejected after 3 days at most. This is despite the P2 team apparently successfully using a cookie they had preserved many months ago. 

To add to the confusion (for me at least) the cookie expiry date is set for many months ahead, not just 3 days. Going off the cookie expiry timestamp, there is no reason for it (cookie) to be expired after merely 3 days.